### PR TITLE
feat: add CORS configuration

### PR DESCRIPTION
**Pull request by zrock-dev**
## Related Tickets & US
- [CS-43-14 implement CORS](https://app.clickup.com/t/868b2p1ek)

## Description

- CORS implementation to allow fronted use our API.
- Implement a custom CORS rule.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

I've manually verified the server accepts the required CORS

```
$ curl -v --request OPTIONS 'http://localhost:8080' -H 'Origin: http://localhost:5173' -H 'Access-Control-Request-Method: GET'
---
Output:
---
* Host localhost:8080 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:8080...
* Connected to localhost (::1) port 8080
* using HTTP/1.x
> OPTIONS / HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/8.10.1
> Accept: */*
> Origin: http://localhost:5173
> Access-Control-Request-Method: GET
>
* Request completely sent off
< HTTP/1.1 200
< Vary: Origin
< Vary: Access-Control-Request-Method
< Vary: Access-Control-Request-Headers
< Access-Control-Allow-Origin: http://localhost:5173
< Access-Control-Allow-Methods: GET,POST,PUT,DELETE
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 0
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Content-Length: 0
< Date: Fri, 29 Nov 2024 03:25:33 GMT
<
* Connection #0 to host localhost left intact

```

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosAPI/22)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -44,10 +44,12 @@ val mockitoDocVersion = "5.5.0"
 dependencies {
     implementation("org.springframework.boot:spring-boot-starter-data-jpa")
     implementation("org.springframework.boot:spring-boot-starter-data-rest")
+    implementation("org.springframework.boot:spring-boot-starter-security")
     implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:$springDocVersion")
     implementation("org.flywaydb:flyway-core")
     implementation("org.flywaydb:flyway-database-postgresql")
+
```
Comment (ronaldo-mendoza-jalafund) [inline build.gradle.kts]: remove the spaces you have added

**Review by ronaldo-mendoza-jalafund — CHANGES_REQUESTED** [review]
Solve the conflicts shown by sonarqube

**Review by teran-joseluis — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (zrock-dev) [reply build.gradle.kts]: Done

**Comment by teran-joseluis** [comment]
It looks good, but just resolve sonarqube issues.

**Review by ronaldo-mendoza-jalafund — APPROVED** [review]
Nice job

**Review by alex-paca-jalafund — APPROVED** [review]
Good work.

**Review by ronaldo-mendoza-jalafund — APPROVED** [review]
nice job

**Pull request merged**